File: java/src/main/java/com/authdog/exceptions/AuthdogException.java
package com.authdog.exceptions;

/**
 * Base exception class for all Authdog SDK errors.
 */
public class AuthdogException extends RuntimeException {

    /**
     * Constructor with message.
     * @param messageParam Error message
     */
    public AuthdogException(final String messageParam) {
        super(messageParam);
    }

    /**
     * Constructor with message and cause.
     * @param messageParam Error message
     * @param causeParam Cause of the exception
     */
    public AuthdogException(final String messageParam,
                            final Throwable causeParam) {
        super(messageParam, causeParam);
    }
}

File: csharp/Exceptions/AuthdogException.cs
using System;

namespace Authdog.Exceptions
{
    /// <summary>
    /// Base exception class for all Authdog SDK errors
    /// </summary>
    public class AuthdogException : Exception
    {
        public AuthdogException(string message) : base(message)
        {
        }

        public AuthdogException(string message, Exception innerException) : base(message, innerException)
        {
        }
    }
}
